src/main/java/com/hierynomus/protocol/commons/buffer/Buffer.java

```diff
@@ -680,7 +680,7 @@ public InputStream asInputStream() {
             @Override
             public int read() throws IOException {
                 try {
-                    return Buffer.this.readByte();
+                    return Buffer.this.readByte() & 0xFF;
```

Comment:
Yes, it is very obscure, it took me some time to get my head around it. When a cast happens from byte to int, if the high bit of the byte is 1 then the int seems to have the upper byte as all 1's. I've added a BufferTest for it.

Btw, Just looked at ByteArrayInputStream source,
    public synchronized int read() {
        return (pos < count) ? (buf[pos++] & 0xff) : -1;
    }